### DISCUSSION TITLE
无网络怎么使用xmake

### DISCUSSION THREAD

**Discussion by wang-jinbo** [xmake-io/xmake#2838] [Q&A]
运行环境几乎无网络。在使用xmake时，添加
```
set_policy("build.c++.modules", true)
add_requires("openmp")
```
后，卡在
`updating repositories .. `
这些包应该在xmake中均有
```
share/xmake/rules/c++/modules
share/xmake/rules/c++/openmp
```
应该不需要额外网络配置

**Comment by waruqi** [comment]
> 这些包应该在xmake中均有
share/xmake/rules/c++/modules
share/xmake/rules/c++/openmp

这不是包，是构建规则，所有的包都在 xmake-repo ，至少需要拉取 xmake-repo 一次，大部分 openmp 配置逻辑都需要包

如果网络不好，就配置代理或者镜像 https://xmake.io/#/zh-cn/package/remote_package?id=%e8%ae%be%e7%bd%ae%e4%bb%a3%e7%90%86

或者自己 clone 下来
